src/expr/src/expr/expr_some_all.rs

```diff
@@ -164,20 +165,20 @@ impl Expression for SomeAllExpression {
         ))
     }
 
-    fn eval_row(&self, row: &OwnedRow) -> Result<Datum> {
-        let datum_left = self.left_expr.eval_row(row)?;
-        let datum_right = self.right_expr.eval_row(row)?;
+    async fn eval_row(&self, row: &OwnedRow) -> Result<Datum> {
+        let datum_left = self.left_expr.eval_row(row).await?;
+        let datum_right = self.right_expr.eval_row(row).await?;
```

Comment:
Not sure whether we should make it concurrent for "left then right"s or "loop"s. But it seems not getting worse considering most of the expressions are CPU bounded.